Source/SwiftLintFramework/Rules/ExplicitTypeInterfaceRule.swift

```diff
@@ -29,8 +43,11 @@ public struct ExplicitTypeInterfaceRule: ASTRule, OptInRule, ConfigurationProvid
                          dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
 
         guard configuration.allowedKinds.contains(kind),
-            !containsType(dictionary: dictionary),
-            let offset = dictionary.offset else {
+            !dictionary.containsType,
+            let offset = dictionary.offset,
+            !file.captureGroupByteRanges.contains(where: { $0.contains(offset) }),
```

Comment:
This will perform the search for the whole file every time 😱 

This is an opt-in rule, so having a worse performance is acceptable, but I'd be awesome if we could think of a way to make it faster.

Performing the regex match just on the parent type would be an option, but it'd have to change a few things. Other option would be performing the whole search first and the navigate the AST manually instead of using `ASTRule`.

Reply:
I choose the later one, thanks for the suggestion. Here is a screenshot of the listing speed from *before the change*:

<img width="687" alt="screen shot 2018-07-28 at 20 29 43" src="https://user-images.githubusercontent.com/6389088/43359964-6281e4c6-92ad-11e8-88f9-ca57b12a97d9.png">